pkg/util/azure/chained_credential.go

```diff
@@ -0,0 +1,113 @@
+package azure
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+)
+
+// This file is a copy of https://github.com/Azure/azure-sdk-for-go/blob/sdk/azidentity/v1.5.1/sdk/azidentity/chained_token_credential.go with a change that
+// removes the specific error checking logic.
+// Velero chains the ConfigCredential, WorkloadIdentityCredential, ManagedIdentityCredential and uses them to do the auth in order.
+// The original ChainedTokenCredential only reports the error got from the last credential and ignores the others in some cases, this causes confusion
+// if the root cause is from the former credentials.
+// For example, if users provide an invalid certificate as the credential, the original reports a managed identity credential error, this is hard to debug.
```

Comment:
Is there a way that we can distinguish auth with Workload Identity and auth with Managed Identity?

Reply:
For example, is there any unique environment variable to identify whether the pod auth with Workload Identity or Managed Identity?

Reply:
valid point, I think for cases when only CLIENT ID is specified, we can be just cautious and print message for BOTH Managed Identity and Workload Identity 
Kind of and edge casee. 
But should be good enough IMO.

Reply:
Currently, we only officially support Workload Identity, so how about removing the Managed Identity? Do you think supporting Managed Identity is a valid use case for Velero?

Reply:
we actually use Managed Identity in downstream because not all customers opt for Workload Identity.
From velero standpoint we try to recommend workload identity as the ideal way.


think of it as a spectrum, generally Service Principal is least preffered.

Now if users go for MSI based, they need to setup workload identity.
managed identity is in between where customer can use MSI without having the cluster admin to support workload identity.

Reply:
We can add docs for Manged Identity if needed. but we can mark that as not-recommended
Users who are just testing out velero can use MI to test without WI